kchd/app.py

```diff
@@ -39,10 +39,11 @@ def _init(self) -> None:
             "status": SystemStatusController(self._mqtt, self.update_leds),
         }
 
-        leds = set().union(*[
-            controller.leds  # type: ignore[attr-defined]
+        leds = {
+            led
             for controller in self._controllers.values()
-        ])
+            for led in controller.leds  # type: ignore[attr-defined]
+        }
```

Comment:
I don't think this is particularly easy to understand.

Equally, the current implementation sucks and this definitely an improvement.